core-api/core_api/routes/chat.py

```diff
@@ -51,6 +51,12 @@ async def semantic_router_to_chain(
 
     selected_chain = None
 
+    def select_chat_chain(chat_request: ChatRequest, routable_chains: dict[str, Runnable]) -> Runnable:
```

Comment:
I think this could live in build_chains or runnables as a RunnableBranch and then be part of the chain itself?